crates/astria-sequencer/src/sequence/action.rs

```diff
@@ -11,8 +11,8 @@ use crate::transaction::action_handler::ActionHandler;
 /// which are bytes to be interpreted by the rollup.
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
 pub struct Action {
-    chain_id: Vec<u8>,
-    data: Vec<u8>,
+    pub chain_id: Vec<u8>,
```

Comment:
Why make this public

Reply:
Not necessarily... But you have to have a good reason to make it all pub.

For example the responses of the json rpc client I made all `pub`, because there aren't really any invariants to uphold (no methods that take them as arguments) and because I didn't care to hide implementation details (it's all Celestia responses anyway).

However, if your `Action` type contains restricted values (because other methods rely on that fact), then you should not allow its construction outside of dedicated methods.

Another reason is not allowing anyone outside your crate to create the type so that the type contents become an implementation details and you don't have to worry about breaking somebody else's code